lib/theory/src/Theory/Tools/Wellformedness.hs

```diff
@@ -408,6 +408,29 @@ factReports thy = concat
           return $ ppFa $-$ text ("show:" ++ show info)
     reservedFactName _ = Nothing
 
+    -- Check for usage of all type facts with reserved names
+    reservedFactNameRules = do
+      ru <- thyProtoRules thy
+      let lfact = [fa| fa <- get rPrems ru
+                      , factTag fa `elem` [KUFact,KDFact] 
+                      || isKLogFact fa]
+          mfact = [fa | fa <- get rActs ru
+                      , factTag fa `elem` [KUFact,KDFact,InFact,OutFact,FreshFact]
+                      || isKLogFact fa]
+          rfact = [fa | fa <- get rConcs ru
+                      , factTag fa `elem` [KUFact, KDFact]
+                      || isKLogFact fa]
+          check _   []  = mzero
+          check msg fas = return $ (,) "Reserved names" $
```

Comment:
Underline as in the other PR?

Reply:
I need the function "underlineTopic" in the [PR](https://github.com/tamarin-prover/tamarin-prover/pull/553) to do this, so I need to wait it to be accepted to merge and then use this function to do underline

Reply:
The other PR is now accepted